connectors/airwatch/src/main/java/com/vmware/connectors/airwatch/AirWatchController.java

```diff
@@ -174,12 +208,116 @@ public AirWatchController(AsyncRestOperations rest, CardTextAccessor cardTextAcc
         return Observable.just(cardBuilder.build());
     }
 
-    private CardAction.Builder getappInstallActionBuilder(String appBundle) {
+    private CardAction.Builder getInstallActionBuilder(String routingPrefix,
+                                                       String appName,
+                                                       String udid,
+                                                       String platform) {
         CardAction.Builder actionBuilder = new CardAction.Builder();
         actionBuilder.setLabel(cardTextAccessor.getActionLabel("installApp"))
-                .setActionKey("INSTALL_APP")
-                .setUrl(cardTextAccessor.getMessage("ws1.installAppUrl", appBundle))
-                .setType(HttpMethod.GET);
+                .setActionKey(CardActionKey.DIRECT)
+                .setUrl(routingPrefix + "/mdm/app/install")
+                .addRequestParam("app_name", appName)
+                .addRequestParam("udid", udid)
+                .addRequestParam("platform", platform)
+                .setType(HttpMethod.POST);
         return actionBuilder;
     }
+
+    private Single<String> getEucToken(URI baseUri, String udid, String platform, String hzn) {
+        logger.trace("getEucToken called: GreenBox base url={}, udid={}, platform={}",
+                gbBaseUri.toString(), udid, platform);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set(COOKIE, "HZN=" + hzn);
+
+        String deviceType = platform.replaceAll("(?i)ios", "Apple");
+
+        ListenableFuture<ResponseEntity<JsonDocument>> future = rest.exchange(
+                "{baseUrl}/catalog-portal/services/auth/eucTokens?deviceUdid={udid}&deviceType={deviceType}",
+                HttpMethod.POST, new HttpEntity<String>(headers), JsonDocument.class,
+                baseUri, udid, deviceType);
+
+        return toSingle(future)
+                .map(entity -> entity.getBody().read("$.eucToken"));
+    }
+
+    private Single<GbConnection> getGbConnection(URI gbBaseUri, String eucToken) {
+        logger.trace("getGbConnection called: GreenBox base url={}", gbBaseUri.toString());
+        return getCsrfToken(gbBaseUri, eucToken)
+                .map(csrfToken -> new GbConnection(gbBaseUri, eucToken, csrfToken));
+    }
+
+    private Single<ResponseEntity<HttpStatus>> installApp(String gbAppName, GbConnection gbSession) {
```

Comment:
If you don't mind, try to name the 2 installApp methods differently.  It's easy to misread the code because of the overloading (especially when it's the same number of parameters and you aren't in an IDE).